x/auth/middleware/gas.go

```diff
@@ -51,6 +51,10 @@ func (txh gasTxHandler) DeliverTx(ctx context.Context, tx sdk.Tx, req abci.Reque
 	}
 
 	res, err := txh.next.DeliverTx(sdk.WrapSDKContext(sdkCtx), tx, req)
+	if err != nil {
+		return abci.ResponseDeliverTx{}, err
+	}
+
```

Comment:
We should actually revert these 4 lines. The gas middleware is set up so that even when there's an error, the `res.GasUsed/Wanted` is set.